### ISSUE TITLE
kontena deploy fails with "not found"

### ISSUE THREAD

**Issue by robxu9** [kontena/kontena#88]
I made a kontena.yml that looks similar to the following:

    # jenkins configuration
    jenkins:
        image: jenkinsci/jenkins:latest
        stateful: true
        instances: 1
        user: jenkins
        ports:
            - "9181:8080"
            - "9150:50000"
        affinity:
            - container!=gitblit

    # gitblit configuration
    gitblit:
        image: elderresearch/docker-gitblit:latest
        stateful: true
        instances: 1
        user: gitblit
        ports:
            - "9180:80"
            - "9118:9418"
            - "9122:29418"
        affinity:
            - container!=jenkins

However, when I run `kontena deploy` with this kontena.yml file, it creates the services then fails with `error: {"error":"Not found"}.`. For some reason it also creates the services as `demo/vagrant-jenkins` and `demo/vagrant-gitblit`. If I call it to deploy manually with `kontena service deploy vagrant-jenkins` and `kontena service deploy vagrant-gitblit`, it works.

Why does it append the `demo/` portion if we're already on the `demo` grid? (Surely that's redundant and confusing, because my first instinct was to do `kontena service deploy demo/vagrant-jenkins`, which doesn't work.) Secondly, it looks that `kontena deploy` is calling the services wrong.

**Comment by nevalla** [comment]
This is regression on 0.7 release and the fix is coming soon.

There are also couple of other thoughts how to fix your kontena.yml file.

I don't know does jenkins service need `user` attribute but at least gitblit does not work if you define it to use gitblit user. So I suggest you to remove it (or use the correct one).

When using `kontena deploy`, kontena prefixes services as you might have noticed. Due that you have to use `prefix` wildcard on affinity rules, for example: `container!=%{prefix}-gitblit-1`. Also notice that you have to define also container number on container affinity.

I got services up and running with the following `kontena.yml` file.
```
# jenkins configuration
jenkins:
    image: jenkinsci/jenkins:latest
    stateful: true
    instances: 1
    user: jenkins
    ports:
        - "9181:8080"
        - "9150:50000"
    affinity:
        - container!=%{prefix}-gitblit-1

# gitblit configuration
gitblit:
    image: elderresearch/docker-gitblit:latest
    instances: 1
    ports:
        - "9180:80"
        - "9118:9418"
        - "9122:29418"
    affinity:
        - container!=%{prefix}-jenkins-1
```

That `demo/` portion on service list is a valid question. I think I will open a new issue where to discuss do we need or should we remove them.

**Comment by jakolehm** [comment]
@robxu9: fix is included in the 0.7.3 release.

**Comment by robxu9** [comment]
Thanks for the advice and the fix!
